src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -96,27 +102,22 @@ private void ForceCompactingGC()
             /// </summary>
             public Stream CreateReadableStream()
             {
-                // CreateViewStream is not guaranteed to be thread-safe
+                // CreateViewAccessor is not guaranteed to be thread-safe
                 lock (_memoryMappedFile)
```

Comment:
You're locking  _memoryMappedFile which is a "ReferenceCountedDisposable".  But i thought everyone got a fresh ReferenceCountedDisposable.  So will this lock actually do anything?